tests/test_main.py

```
@@ -3144,15 +3144,38 @@ async def mock_connect_matrix_fn(*args, **kwargs):
+        def mock_check_conn() -> object:
+            # Return a non-coroutine sentinel; create_task is patched below.
+            """
+            Provide the sentinel value representing a mocked connection check.
+
+            Returns:
+                object: The sentinel `check_conn_sentinel` used by the patched `asyncio.create_task` to identify the mocked check-connection invocation.
+            """
+            return check_conn_sentinel
 
-        def mock_create_task(coro, *args, **kwargs):
+        def mock_create_task(coro: Any, *_args: Any, **_kwargs: Any) -> Any:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `coro`

(ANN401)